File: jarvis/221007/두수의합.py
def solution(queue1, queue2):
    answer = 0
    sum_item = sum(queue1) + sum(queue2)
    queue3 = queue1 + queue2
    cur_sum = sum(queue1)
    
    if sum_item%2 != 0 :
        return -1

    target_value = sum_item // 2

    start = 0
    end = len(queue1)-1

    while target_value != cur_sum:
        if cur_sum > target_value:
            cur_sum -= queue3[start]
            start += 1
            answer += 1
            if start > end or start > len(queue3) : 
                return -1
        else:
            end += 1
            answer += 1
            if end == len(queue3):
                return -1 
            cur_sum += queue3[end]

    return answer

print(solution([1, 2, 1, 2], [1, 10, 1, 2]))

File: jarvis/221007/두수의합.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> queue1, vector<int> queue2) {
    long long answer = 0;
    long long sum1 = 0, sum2 = 0;
    int k = queue1.size() * 2;

    for (int i = 0; i < queue1.size(); i++) {
        sum1 += queue1[i];
    }
    for (int i = 0; i < queue2.size(); i++) {
        sum2 += queue2[i];
    }

    while (k--) {
        if (sum1 > sum2) {
            sum1 -= queue1.front();
            sum2 += queue1.front();
            queue2.push_back(queue1.front());
            queue1.erase(queue1.begin());
            answer++;
        }
        else if (sum1 < sum2) {
            sum2 -= queue2.front();
            sum1 += queue2.front();
            queue1.push_back(queue2.front());
            queue2.erase(queue2.begin());
            answer++;
        }
        else
            break;
    }
    if (sum1 != sum2) {
        return -1;
    }
    return answer;
}

int main(void){
    vector<int> v;
    v.push_back(3);
    v.push_back(3);
    v.push_back(3);
    v.push_back(3);

    vector<int> v2;
    v2.push_back(3);
    v2.push_back(3);
    v2.push_back(21);
    v2.push_back(3);
    int answer = solution(v, v2);
    cout << answer;
    return 0;
}